### PR TITLE
Inputchecks

### PR DESCRIPTION
**Pull request by etalbert102**
added input checks for each geostructure

### CODE REVIEW

File: geostructures/geohash.py [hunk 1]
```diff
@@ -629,3 +631,27 @@ def hash_shape(self, shape: GeoShape, **_):
             return self._hash_linestring(shape)
 
         return self._hash_polygon(shape)
+
+
+def convert_hashmap(hexmap: Dict[str, float]):
```
Comment (ccbest) [inline geostructures/geohash.py:636]: 1 branch for one thing - let's move this function to another PR

File: geostructures/structures.py [hunk 2]
```diff
@@ -959,6 +961,11 @@ def __init__(
         self.nw_bound = nw_bound
         self.se_bound = se_bound
 
+        if not isinstance(nw_bound, Coordinate):
```
Comment (ccbest) [inline geostructures/structures.py:964]: Is there a functional way to abstract some of this? Like if you had a function along the lines of:
```
_ensure_types([nw_bound, se_bound], [Coordinate, Coordinate])
```
It would probably reduce the amount of code needed to check each individual type

File: geostructures/structures.py [hunk 3]
```diff
@@ -1320,11 +1342,21 @@ def __init__(  # pylint: disable=too-many-arguments
...
+            self.inner_radius = float(inner_radius)
+            self.outer_radius = float(outer_radius)
+        except ValueError:
+            raise ValueError("radius distance should be in meters")
         self.angle_min = angle_min
         self.angle_max = angle_max
 
+        if not isinstance(center, Coordinate):
+            raise Exception('The center must be a coordinate. The input value was: {}'.format(center))
+        if angle_min < 0 or angle_min > 360:
```
Comment (ccbest) [inline geostructures/structures.py:1355]: Why does it have to be bound to 0 <= angle_min <= 360? Shouldn't -70 be the same as 290?

File: geostructures/structures.py [hunk 4]
```diff
@@ -1515,6 +1548,9 @@ def __init__(
         super().__init__(dt=dt, properties=properties)
         self.coords = coords
 
+        if not isinstance(coords[0], Coordinate):
+            raise Exception('The list must contain coordinates. The inputted value was: {}'.format(coords))
```
Comment (ccbest) [inline geostructures/structures.py:1552]: Use f-string instead of `.format`

**Review by ccbest — CHANGES_REQUESTED** [review]
Overall good work and coverage over the changes needed, but lets see if we can refactor/simplify it a little

**Comment by etalbert102** [comment]
Input checks add too much computational time
